plugins/scheduled_task/lib/object_task_engine/KObjectTaskModifyEntryEngine.php

```diff
@@ -0,0 +1,225 @@
+			
+			$entryObj = $this->updateEntryFromMetadata($metadataPlugin, $inputMetadataArr, $entryObj, $metadataFilter);
+		}
+
+		if(is_null($entryObj->userId))
+			$entryObj->userId = $objectTask->inputUserId;
+		if(is_null($entryObj->entitledUsersEdit))
+			$entryObj->entitledUsersEdit = $objectTask->inputEntitledUsersEdit;
+		if(is_null($entryObj->entitledUsersPublish))
+			$entryObj->entitledUsersPublish = $objectTask->inputEntitledUsersPublish;
+
+		try
+		{
+			$client->baseEntry->update($entryId, $entryObj);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with entry update entry id $entryId - " . $e->getMessage());
+		}
+		KBatchBase::unimpersonate();
+	}
+	
+	private function updateMetadataObj(KalturaBaseEntry $entryResultForMetadataUpdate, &$metadataPlugin, $outputMetadataArr, $outputMetadataProfileId, $metadataFilter)
+	{
+		$entryId = $entryResultForMetadataUpdate->id;
+		
+		try
+		{
+			$metadataInputResult = $metadataPlugin->metadata->listAction($metadataFilter, null);
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::notice("problem with metadata list with entry id $entryId - " . $e->getMessage());
+		}
+		
+		$tepmlateXmlObj = $this->getMetadataXmlTemplate($metadataPlugin, $outputMetadataProfileId);
+	
+		if($metadataInputResult && $metadataInputResult->totalCount != 0)
+		{
+			$metadataId = $metadataInputResult->objects[0]->id;
+			$metadataInputXmlStr = $metadataInputResult->objects[0]->xml;
+			$currentXmlObj = new SimpleXMLElement($metadataInputXmlStr);
+			
+			if(!is_null($tepmlateXmlObj))
+			{
+				$xmlData = $this->getUpdatedMetadataXmlStrFromEntry($entryResultForMetadataUpdate, $tepmlateXmlObj, $currentXmlObj, $outputMetadataArr);
+			
+				try
+				{
+					$metadataPlugin->metadata->update($metadataId, $xmlData, null);
+				}
+				catch(Exception $e)
+				{
+					KalturaLog::notice("problem with metadata update entry id $entryId - " . $e->getMessage());
+				}
+			}
+		}
+		else
+		{
+			
+			$xmlObj = new SimpleXMLElement("<metadata></metadata>");
+			$xmlData = $this->getUpdatedMetadataXmlStrFromEntry($entryResultForMetadataUpdate, $tepmlateXmlObj, $xmlObj, $outputMetadataArr);
+	
+			try
+			{
+				$metadataPlugin->metadata->add($outputMetadataProfileId, KalturaMetadataObjectType::ENTRY, $entryId, $xmlData);
+			}
+			catch(Exception $e)
+			{
+				KalturaLog::notice("problem with metadata add entry id $entryId - " . $e->getMessage());
+			}
+		}
+	}
+	
+	private function getUpdatedMetadataXmlStrFromEntry(KalturaBaseEntry $entryResultForMetadataUpdate, SimpleXMLElement $emptyXmlObj, SimpleXMLElement $currentXmlObj, array $outputMetadataArr)
+	{
+		$tempXmlObj = $emptyXmlObj;		
+		
+		KalturaLog::debug("current xml object - " . print_r($currentXmlObj, true));
+		KalturaLog::debug("output metadata array - " . print_r($outputMetadataArr, true));
+		
+		foreach($tempXmlObj as $metadataFieldName => $emptyXmlObjItem)
+		{
+			foreach($currentXmlObj as $currMetadataFieldName => $currentXmlObjItem)
```

Comment:
Instead of the internal loop , check if value exist in the currentXmlObjItem mand it is not empty